roles/irods/tasks/icat.yml

```diff
@@ -1,29 +1,167 @@
-#installing the iRODS ICAT
+# installing the iRODS ICAT
+- name: Create {{ irods_service_account }} group
+  group:
+    name: "{{ irods_service_account }}"
+    state: present
+  become: true
+
+- name: Creating {{ irods_service_account }} user
+  user:
+    name: "{{ irods_service_account }}"
+    groups: "{{ irods_service_account }},admin"
+  become: true
+
 - name: Installing the iRODS Package(s)
-  become: True
-  yum: 
-    name: "{{ item }}"
+  yum:
+    name: '{{ item }}'
     state: present
   loop:
-  - irods-server
-  - irods-database-plugin-postgres
+    - irods-server
+    - irods-database-plugin-postgres
+    - irods-resource-plugin-s3-4.2.10.0-1
+    - irods-rule-engine-plugin-python
+  become: true
+
+- name: Make sure that the hostname is set correct FQDN name
+  ansible.builtin.hostname:
+    name: '{{ irods_icat_fqdn }}'
+  become: true
+
+- name: Ensure that /etc/irods directory exists and is owned by {{ irods_service_account }}
+  ansible.builtin.file:
+    path: '/etc/irods'
+    state: directory
+    owner: '{{ irods_service_account }}'
+    group: '{{ irods_service_account }}'
+    mode: '0755'
+  become: true
 
-- name: set hostname to DNS.
-  become: True
-  hostname:
-    name: "{{ ip_addresses[inventory_hostname].fqdn }}"
+- name: Install certificate key and chain files for iRODS
+  ansible.builtin.copy:
+    src: 'files/{{ slurm_cluster_name }}_cluster/nemi_irods/{{ item.name }}'
+    dest: '/etc/irods/{{ item.name }}'
+    owner: '{{ irods_service_account }}'
+    group: '{{ irods_service_account }}'
+    mode: '{{ item.mode }}'
+    decrypt: true
+    force: true
+  with_items:
+    - name: '{{ irods_ssl_certificate_key_file }}'
+      mode: '0600'
+    - name: '{{ irods_ssl_certificate_chain_file }}'
+      mode: '0644'
+  become: true
+
+- name: If missing, build 4096 bit DHparam /etc/irods/{{ irods_ssl_dh_params_file }} file (takes several minues ...)
+  ansible.builtin.command:
+    cmd: /bin/openssl dhparam -2 -out /etc/irods/{{ irods_ssl_dh_params_file }} 4096
+    creates: /etc/irods/{{ irods_ssl_dh_params_file }}
+  register: result
+  failed_when:
+    - result.rc != 0
+  become: true
+
+- name: Set ownership of {{ irods_ssl_dh_params_file }}
+  ansible.builtin.file:
+    path: '/etc/irods/{{ irods_ssl_dh_params_file }}'
+    owner: '{{ irods_service_account }}'
+    group: '{{ irods_service_account }}'
+    mode: '0644'
+  become: true
 
 - name: create_unattended_installation_json
-  become: True
   template:
     src: unattended_install.json.j2
     dest: /root/unattended_install.json
+    mode: '0600'
   register: create_unattended_installation_json
+  become: true
+
+# - name: Check if {{ irods_db_name }} database is available
```

Comment:
Do we need the commented code or can this be deleted?

Reply:
Was there, as this part was only recently changed. And was for the potential fallback. Removed now.